### ISSUE TITLE
Work on memory usage

### ISSUE THREAD

**Issue by alexhultman** [seemk/WebUDP#2]
I'm interested in seeing how tight a WebRTC connection can be in memory usage. I did a simple test of the current code, 14 tabs opened against the EchoServer and saw significant memory usage in the server. It jumps from ~500kb to over 1 MB per 14 tabs, you think this can be improved? What consumes memory? DTLS? If you can get down to a very slim memory usage, like only a few bytes per connection, that would be *really* interesting.

**Comment by seemk** [comment]
I haven't measured the memory usage yet, however I am suspecting that most of the memory per client is indeed taken up by the SSL context. The server preallocates memory for a fixed number of clients (excluding SSL context) at the moment, should probably fix this.

**Comment by alexhultman** [comment]
The mandatory DTLS is a real tearjerker. 32kb * 14 = ~500kb, maybe it really does require 32kb per connection? Options to tweak?

With UDP below unordered SCTP you have the *opportunity* to go below what is needed in memory for TCP (at least what is needed by most TCP stacks), but that seems to depend strongly on how heavy this darn DTLS is.

I was investigating doing something similar as this project myself, but when I saw this mandatory DTLS I just dropped it. Will be interesting to follow this project and see if it is possible to improve from here :sparkles:
